server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -230,4 +230,46 @@ protected void checkDuplicates(
     updatedDuplicateOptions.putAll(customDuplicateOptions);
     checkDuplicateEntries(ctx, foundEntries, updatedDuplicateOptions);
   }
+
+  /**
+   * Flags errors for rule lists passed as parameters if there are multiple instances of mutually
+   * exclusive options.
+   *
+   * @param options Options checked to insert into error message
+   * @param rules Generic lists of rule lists to check. Will be a collection of ParserRuleContext
+   *     and/or TerminalNode objects.
+   * @param <E> List of Generics to allow cross-rule context collection.
+   * @return True if an instance is not found
+   */
+  protected <E extends ParseTree> boolean checkHasMutuallyExclusiveOptions(
+      String options, List<E>... rules) {
+    String token = null;
+    for (List<E> option : rules) {
+      if (option.isEmpty()) continue;
+      if (token == null) token = option.remove(0).getText().split("\\(")[0].toUpperCase();
```

Comment:
The `remove` call modifies the list, right?
Wouldn't `getClass` provide the needed information to determine mixing of alternatives?

Reply:
The problem is that e.g. a valid input will be modified, so any additional processing after this step will not see the node.